### PR TITLE
[feat]: index_fill_ frontend pytorch op

### PR DESCRIPTION
**Pull request by cocoshe**
### Details:

Add support for aten::index_fill_ for pytorch models

### Tickets:

[#23326](https://github.com/openvinotoolkit/openvino/issues/23326)

### CODE REVIEW

File: src/frontends/pytorch/src/op/index_fill_.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    auto value = context.get_input(3);
+
+    auto const_1_vec = v0::Constant::create(element::i32, Shape{1}, {1});
+
+    auto tensor_rank = std::get<1>(get_shape_rank(context, input, true));
+    auto tensor_rank_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(tensor_rank, dim));
+    auto positive_dim = normalize_axis(context, dim, tensor_rank_correct_type);
+
+    // scalar to vec
+    auto dim_vec = context.mark_node(std::make_shared<v1::Reshape>(positive_dim, const_1_vec, false));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/index_fill_.cpp:42]:
```suggestion
    auto tensor_rank = std::get<1>(get_shape_rank(context, input, false));
    auto tensor_rank_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(tensor_rank, dim));
    auto dim_vec = normalize_axis(context, dim, tensor_rank_correct_type);

    // scalar to vec
```
You can skip `Reshape` to 1D if you get rank as 1D

**Review by mvafin — CHANGES_REQUESTED** [review]
In general looks good, please also fix code style

**Comment by cocoshe** [comment]
> In general looks good, please also fix code style

OK, and by the way, the code style check can only be enabled when CMake? Some other projects support the code style check by  `pre-commit`, does openvino support it?

**Comment by mlukasze** [comment]
yes, that's how we know it should be corrected for this PR ;)

**Comment by cocoshe** [comment]
> yes, that's how we know it should be corrected for this PR ;)

I'm not very sure about it, the CI use clang-format tool here:
https://github.com/openvinotoolkit/openvino/blob/3b596084d35b5dd4100e14feaf70b95b699cd51b/.github/workflows/code_style.yml#L30-L31

I used to participate in some projects like `PaddlePaddle`, it support the [`pre-commit`](https://pre-commit.com/#intro) with the `.pre-commit-config.yaml` config file:

https://github.com/PaddlePaddle/Paddle/blob/c3ba53c8c9df394d2054e5d1e19eab357c9b6b26/.pre-commit-config.yaml#L68-L75

```yaml
# For C++ files
-   repo: local
    hooks:
    -   id: clang-format
        name: clang-format
        description: Format files with ClangFormat.
        entry: bash ./tools/codestyle/clang_format.sh -i
        language: system
        files: \.(c|cc|cxx|cpp|cu|h|hpp|hxx|xpu|kps)$
```

It seems the CI requires the `make` process, while `PaddlePaddle` can run the code style check when `git commit -m 'xxxx'`, it only focus on the file you modified, and don't need to run `make` again. With the `pre-commit` hook, just run `git commit -m "xxxx"` can get the modified result directly, and all we need is to run `git add` and `git commit` again to add the fixed code. `PaddlePaddle` feels more convenient since the `pre-commit` don't need to `make` the code style check again and again. And the CI just need to [run the bash](https://github.com/PaddlePaddle/Paddle/blob/c3ba53c8c9df394d2054e5d1e19eab357c9b6b26/tools/codestyle/pre_commit.sh) to show "It is a code style problem, and where is the problem, and how to fix the code style problem":

```
echo -e '\n************************************************************************************'
if [ ${check_error} != 0 ];then
    echo "Your PR code style check failed."
    echo "Please install pre-commit locally and set up git hook scripts:"
    echo ""
    echo "    pip install pre-commit==2.17.0"
    echo "    pre-commit install"
    echo ""
    if [[ $num_diff_files -le 100 ]];then
        echo "Then, run pre-commit to check codestyle issues in your PR:"
        echo ""
        echo "    pre-commit run --files" $(echo ${diff_files} | tr "\n" " ")
        echo ""
    fi
    echo "For more information, please refer to our codestyle check guide:"
    echo "https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/git_guides/codestyle_check_guide_cn.html"
else
    echo "Your PR code style check passed."
fi
echo -e '************************************************************************************\n'
```
So I'm not sure if the the `openvino` only support the code style check with `make`. If yes, I think a `pre-commit` like `PaddlePaddle` did maybe much more convenient.

Just a few feelings when contributing code for `openvino` 👀💭

**Review by mvafin — APPROVED** [review]
Looks good, tests are passing

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — CHANGES_REQUESTED** [review]
Was wrong. Tests fail on windows:
```
RuntimeError: index_fill_(): Expected dtype int64 for index.
```

**Comment by cocoshe** [comment]
> Was wrong. Tests fail on windows:
> 
> ```
> RuntimeError: index_fill_(): Expected dtype int64 for index.
> ```

I don't have any win device for now, and not very sure about what makes the error only on win but not on linux.
The error occured when generating torchscript, and I just noticed some other tests define the index dtype as `long` specifically for torchscript generation. So I commit a bit on it, could you pls rerun the CI for me? Or any other advice or help on solving the error?

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — APPROVED** [review]

**Comment by cocoshe** [comment]
@mvafin Hi, Anything else I should do?

**Comment by mvafin** [comment]
> @mvafin Hi, Anything else I should do?

@cocoshe No, I will take it from here

**Comment by mvafin** [comment]
build_jenkins

**Pull request merged**